Compiling for Windows fails (mingw-w64-gcc)

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
master (https://github.com/godotengine/godot/commit/886afa9b7692bc263cf96c03882760521b74b942)
deleted godot folder and cloned again


**OS/device including version:**
Antergos (Arch Linux)


**Issue description:**
Output:
```Swift

[Initial build] Linking Program        ==> bin/godot.windows.opt.tools.64.exe
[Initial build] /usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Music[.refptr.FOLDERID_Music]+0x0): undefined reference to `FOLDERID_Music'
/usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Videos[.refptr.FOLDERID_Videos]+0x0): undefined reference to `FOLDERID_Videos'
/usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Downloads[.refptr.FOLDERID_Downloads]+0x0): undefined reference to `FOLDERID_Downloads'
/usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Documents[.refptr.FOLDERID_Documents]+0x0): undefined reference to `FOLDERID_Documents'
/usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Pictures[.refptr.FOLDERID_Pictures]+0x0): undefined reference to `FOLDERID_Pictures'
/usr/lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld: platform/windows/os_windows.windows.opt.tools.64.o:os_windows.cpp:(.rdata$.refptr.FOLDERID_Desktop[.refptr.FOLDERID_Desktop]+0x0): undefined reference to `FOLDERID_Desktop'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.windows.opt.tools.64.exe] Error 1
scons: building terminated because of errors.

```


**Steps to reproduce:**
```Swift
git clone https://github.com/godotengine/godot.git
scons p=windows bits=64 target=release_debug tools=yes -j 8
```

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -330,7 +330,7 @@ def configure_mingw(env):
     env.Append(CCFLAGS=['-DWASAPI_ENABLED'])
     env.Append(CCFLAGS=['-DWINMIDI_ENABLED'])
     env.Append(CCFLAGS=['-DWINVER=%s' % env['target_win_version'], '-D_WIN32_WINNT=%s' % env['target_win_version']])
-    env.Append(LIBS=['mingw32', 'opengl32', 'dsound', 'ole32', 'd3d9', 'winmm', 'gdi32', 'iphlpapi', 'shlwapi', 'wsock32', 'ws2_32', 'kernel32', 'oleaut32', 'dinput8', 'dxguid', 'ksuser', 'imm32', 'bcrypt','avrt'])
+    env.Append(LIBS=['mingw32', 'opengl32', 'dsound', 'ole32', 'd3d9', 'winmm', 'gdi32', 'iphlpapi', 'shlwapi', 'wsock32', 'ws2_32', 'kernel32', 'oleaut32', 'dinput8', 'dxguid', 'ksuser', 'imm32', 'bcrypt', 'avrt', 'uuid'])
 
     env.Append(CPPFLAGS=['-DMINGW_ENABLED'])
 

--- a/platform/windows/os_windows.cpp
+++ b/platform/windows/os_windows.cpp
@@ -53,6 +53,7 @@
 
 #include <avrt.h>
 #include <direct.h>
+#include <knownfolders.h>
 #include <process.h>
 #include <regstr.h>
 #include <shlobj.h>

